vmmlib/vector.hpp

```diff
@@ -324,57 +303,19 @@ template< size_t M, typename T > class vector
         return os;
     }
 
-    T array[ M ];    //!< storage
-
-#ifndef SWIG
-    // Vector3 defaults
-    static const vector FORWARD;
-    static const vector BACKWARD;
-    static const vector UP;
-    static const vector DOWN;
-    static const vector LEFT;
-    static const vector RIGHT;
-
-    static const vector ONE;
-    static const vector ZERO;
-
-    // Unit vectors
-    static const vector UNIT_X;
-    static const vector UNIT_Y;
-    static const vector UNIT_Z;
-#endif
-
-}; // class vector
+    /** @name Convenience presets for 3 and 4 component vectors */
+    //@{
+    static vector< M, T > zero();
```

Comment:
We've replaced them to get rid of static initializer ordering issues. Client code looks marginally different (vmml::Vector3f::forward() instead of vmml::Vector3f::FORWARD)